extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -51,49 +51,71 @@ export class BookTocManager implements IBookTocManager {
 		this.targetBookContentPath = targetBook?.bookItems[0].rootContentPath;
 	}
 
-	async getAllFiles(toc: JupyterBookSection[], directory: string, filesInDir: string[], rootDirectory: string): Promise<JupyterBookSection[]> {
-		await Promise.all(filesInDir.map(async file => {
-			let isDirectory = (await fs.promises.stat(path.join(directory, file))).isDirectory();
-			if (isDirectory) {
-				let files = await fs.promises.readdir(path.join(directory, file));
-				let initFile: string = '';
-				//Add files named as readme or index within the directory as the first file of the section.
-				files.some((f, index) => {
-					if (initMarkdown.includes(f)) {
-						initFile = path.parse(f).name;
-						files.splice(index, 1);
-					}
-				});
-				let jupyterSection: JupyterBookSection = {
-					title: file,
-					file: path.join(file, initFile),
-					expand_sections: true,
-					numbered: false,
-					sections: []
-				};
-				toc.push(jupyterSection);
-				await this.getAllFiles(toc, path.join(directory, file), files, rootDirectory);
-			} else if (allowedFileExtensions.includes(path.extname(file))) {
-				// if the file is in the book root we don't include the directory.
-				const filePath = directory === rootDirectory ? path.parse(file).name : path.join(path.basename(directory), path.parse(file).name);
-				const addFile: JupyterBookSection = {
-					title: path.parse(file).name,
-					file: filePath
+	/**
+	 * Replicates Jupyter Book logic to add the init markdown files
+	 * at the beginning of the table of contents.
+	 * @param files The files in the directory
+	*/
+	getInitFile(files: string[]): path.ParsedPath | undefined {
+		let initFile = undefined;
+		// Add files named as readme or index within the directory as the first file of the section.
```

Comment:
This comment doesn't seem to match what the logic is doing - it's not "adding" anything. It's just finding the first file that is included in the initMarkdown. And actually even that logic doesn't seem correct - it's using some but it's not returning true or false in the callback so it'll always go through all the files (so it's actually going to find the last init file)

The function comment has the same issue too